test/e2e/plugin_lifecycle_test.go

```diff
@@ -23,126 +25,200 @@ import (
+				},
+			}
+			test.EventuallyCreated(test.Ctx, test.K8sClient, cluster)
+
+			pluginDefinitionList := &greenhousev1alpha1.PluginDefinitionList{}
+			pluginList := &greenhousev1alpha1.PluginList{}
+			podList := &v1.PodList{}
+			ctx := test.Ctx
+
+			// Creating plugin definition
+			testPluginDefinition := fixtures.CreateTestHookPluginDefinition(test.Ctx, setup)
+			err := test.K8sClient.List(ctx, pluginDefinitionList)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(pluginDefinitionList.Items)).To(BeEquivalentTo(1))
+
+			// Creating plugin
+			_ = setup.CreatePlugin(test.Ctx, "test-hook-plugin",
+				test.WithPluginDefinition(testPluginDefinition.Name),
+				test.WithCluster(secret.Name),
+				test.WithReleaseNamespace(setup.Namespace()))
+
+			// Check jobs
+			jobList := &batchv1.JobList{}
+			err = remoteClient.List(ctx, jobList, client.InNamespace(setup.Namespace()))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(jobList.Items)).To(BeEquivalentTo(0))
+
+			// Check plugin list
+			var plugin greenhousev1alpha1.Plugin
+			Eventually(func(g Gomega) {
+				err = test.K8sClient.List(ctx, pluginList)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
+				g.Expect(pluginList.Items[0].Status.HelmReleaseStatus).ToNot(BeNil())
+				g.Expect(pluginList.Items[0].Status.HelmReleaseStatus.Status).To(BeEquivalentTo("deployed"))
+				plugin = pluginList.Items[0]
+			}).Should(Succeed())
+
+			// Checking pod
+			err = remoteClient.List(ctx, podList, client.InNamespace(setup.Namespace()))
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func(g Gomega) {
+				err = remoteClient.List(ctx, podList, client.InNamespace(setup.Namespace()))
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(podList.Items).ToNot(BeEmpty())
+			}).Should(Succeed())
+
+			// Checking the name of pod
+			podExists := false
+			for _, pod := range podList.Items {
+				if strings.Contains(pod.Name, "alpine") {
+					podExists = true
+					break
+				}
+			}
+			Expect(podExists).To(BeTrue())
 
-		// Check, is deployment deleted
-		Eventually(func(g Gomega) bool {
-			err = remoteClient.List(ctx, deploymentList, client.InNamespace(setup.Namespace()))
-			g.Expect(err).NotTo(HaveOccurred())
-			g.Expect(len(deploymentList.Items)).To(BeEquivalentTo(0))
-			return true
-		}).Should(BeTrue())
+			// Update plugin value
+			setOptionValueForPlugin(&plugin, "hook_enabled", "true")
+			err = test.K8sClient.Update(ctx, &plugin)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Check jobs
+			Eventually(func(g Gomega) {
+				err = remoteClient.List(ctx, jobList, client.InNamespace(setup.Namespace()))
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(len(jobList.Items)).To(BeEquivalentTo(1))
+			}).Should(Succeed())
 
-		// Deleting plugin definition
-		test.EventuallyDeleted(ctx, test.K8sClient, testPluginDefinition)
+		})
 	})
 })
+
+func setOptionValueForPlugin(plugin *greenhousev1alpha1.Plugin, key, value string) {
```

Comment:
Instead of having a helper directly here in the test suite, could you add it analogous to this [one](https://github.com/cloudoperators/greenhouse/blob/747708b779a5bcfa0c603481611b4ff30d653c1d/pkg/test/resources.go#L214-L235) in the `resources.go`?